From dfe52b0831f536157e2547c1d27b249f5d3ac684 Mon Sep 17 00:00:00 2001
From: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>
Date: Wed, 10 Dec 2025 14:27:03 +0530
Subject: [PATCH] add xfce-settings and wrapper to .desktop file

XFCE-Settings categories are necessary for the "calibrate touchscreen" entry to
appear in XFCE's settings dialog. They are necessary here for NI's Linux RT gui.

The Exec line is changed to point to the wrapper script NI uses. This has to be
done in a patch because it needs to be applied before xinput-calibrator
installs the .desktop file to disk.

Upstream-Status: Inappropriate [NI specific]

Signed-off-by: Alex Stewart <alex.stewart@ni.com>
[shubhanshumt26: Regenerated patch.]
Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>
---
 scripts/xinput_calibrator.desktop | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/xinput_calibrator.desktop b/scripts/xinput_calibrator.desktop
index 1d48dab..8e56a9a 100644
--- a/scripts/xinput_calibrator.desktop
+++ b/scripts/xinput_calibrator.desktop
@@ -5,9 +5,9 @@ Name[nl]=Touchscreen Kalibreren
 Name[ru]=Калибровка тачскрина
 Comment=Run the calibration tool for touchscreens
 Comment[ru]=Запуск утилиты калибровки сенсорного экрана
-Exec=/bin/sh -c "xinput_calibrator; cat"
+Exec=/usr/bin/xinput_calibrator_wrapper
 Terminal=true
 Type=Application
 Icon=xinput_calibrator
 StartupNotify=true
-Categories=System;Settings;
+Categories=System;Settings;X-XFCE-SettingsDialog;X-XFCE-HardwareSettings;
